### PR TITLE
Add db/id special case to List/Watch Index

### PR DESCRIPTION
**Pull request by evanphx**
## Summary by CodeRabbit

- **New Features**
  - Enhanced support for listing and watching entities by their unique ID, enabling direct queries and real-time updates using the ID attribute.

- **Tests**
  - Added tests to verify listing and watching entities by ID, covering edge cases and event notifications.

### CODE REVIEW

File: pkg/entity/store.go [hunk 1]
```diff
@@ -661,6 +662,25 @@ func (s *EtcdStore) deleteFromCollection(entity *Entity, collection string) erro
...
+			return nil, err
+		}
+
+		if gr.Count == 0 {
+			return nil, nil
+		}
+
+		return []Id{attr.Value.Id()}, nil
+	}
+
```
[the remark this answers is not in the document]
Comment (evanphx) [reply pkg/entity/store.go:683]: We feel that returning nil for empty slices is fine. It doesn't break anything about the API surface area.

File: pkg/entity/store.go [hunk 2]
```diff
@@ -687,6 +707,46 @@
...
+								},
+							},
+						},
+					}
+				}
+			}
+		}()
+
+		return ret, nil
+	}
```
[the remark this answers is not in the document]
Comment (evanphx) [reply pkg/entity/store.go]: This kludge in temporary, but you rightly point out that we need to close the ret channel, I've done that.

File: pkg/entity/store.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (evanphx) [reply pkg/entity/store.go:683]: Using s.client.Get is preferred. Path sanitization is not needed, ids are hashed to build keys.

File: pkg/entity/store_test.go [hunk 3]
```diff
@@ -521,6 +521,27 @@ func TestEtcdStore_ListIndex(t *testing.T) {
...
+			wantErr:   true,
+		},
+		{
+			name:      "indexed by dbid",
+			attrID:    DBId,
+			value:     RefValue("test1"),
+			wantCount: 1,
+			wantErr:   false,
+		},
+		{
```
[the remark this answers is not in the document]
Comment (evanphx) [reply pkg/entity/store_test.go:538]: Yes, but we provided an Ident for the entity, which is lifted into the ID field.

File: pkg/entity/store_test.go [hunk 4]
```diff
@@ -536,3 +557,159 @@
...
+				t.Errorf("Expected PUT event, got %v", event.Type)
+			}
+			if string(event.Kv.Value) != string(entity1.ID) {
+				t.Errorf("Expected entity ID %s, got %s", entity1.ID, event.Kv.Value)
+			}
+		}
+	case <-time.After(time.Second):
+		t.Fatal("Timeout waiting for creation event")
+	}
+}
```
[the remark this answers is not in the document]
Comment (evanphx) [reply pkg/entity/store_test.go:715]: Same as the previous comment, CreateEntity lifts the value from Ident into ID, so this works as coded.
